internal/currency/usd.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package currency
+
+import (
+	"fmt"
+)
+
+// USD describes USD currency.
+type USD struct {
+	cents int
```

Comment:
I do find this odd

Reply:
You can take a look at https://golang.org/src/time/time.go?s=6278:7279#L117 for something similar.

Reply:
> The USD (United States Dollar) is the official currency of the United States of America. The United States dollar, or U.S. dollar, is made up of 100 cents. It is represented by the symbol $ or US$ to differentiate it from other dollar-based currencies.

I think this is where the confusion comes from. `USD` is the official term for `Dollars`. If I understand correctly, I think the type here could be `Unit` ?
```golang
type Unit struct {
   cents int
}
```

Reply:
I think time is different than currency. Not worth changing but is definitely odd

Reply:
The standard unit for time is seconds, however it's represented as combination of seconds and nanoseconds. So I'm not seeing a large difference in that.

The `type Cents int` has the problem that it can be misconverted easily and miscomputed.

Reply:
I think maybe you're confusing the similarity of time and currency USD is a specific type of currency

Reply:
It's just an odd naming convention to call something USD but it be Cents

Reply:
I see your point it's just odd. I'm not great with names so will leave that to you but it is odd.

Reply:
I think this comes from the difference of the "concept what it is" and "how it is implemented". `cents` is completely an implementation detail to avoid floating point errors with calculations. The concept is some amount of currency in USD.

e.g. it also could be represented like:
```
package money

type Currency struct {
	Name string
	Short string
	Symbol string
}

var USD = Currency{"United States dollar", "USD", "$"}

type Amount struct {
	Value    Fixed100
	Currency Currency
}

type Fixed100 int64
```